### PR TITLE
move dialogue token injection to i18next

### PR DESCRIPTION
**Pull request by ben-lear**
https://github.com/user-attachments/assets/234355d1-9df2-4a4e-bebf-1c32e6f76b03

### CODE REVIEW

File: src/data/mystery-encounters/mystery-encounter-utils.ts [hunk 1]
```diff
@@ -174,30 +176,28 @@ export function getEncounterText(scene: BattleScene, textKey: TemplateStringsArr
...
-    });
-  }
-
-  return textString;
+  // const dialogueTokens = scene.currentBattle?.mysteryEncounter?.dialogueTokens;
+  // if (dialogueTokens) {
+  //   dialogueTokens.forEach((value) => {
+  //     textString = textString.replace(value[0], value[1]);
+  //   });
+  // }
```
Comment (flx-sta) [inline src/data/mystery-encounters/mystery-encounter-utils.ts:190]: no leftovers please

File: src/data/mystery-encounters/mystery-encounter.ts [hunk 2]
```diff
@@ -95,7 +95,7 @@ export default interface IMysteryEncounter {
    * Can be set for uses programatic dialogue during an encounter (storing the name of one of the party's pokemon, etc.)
    * Example use: see MYSTERIOUS_CHEST
    */
-  dialogueTokens?: Map<string, [RegExp, string]>;
+  dialogueTokens?: { [key: string]: string; };
```
Comment (flx-sta) [inline src/data/mystery-encounters/mystery-encounter.ts:98]:
```suggestion
  dialogueTokens?: Record<string, string>;
```

File: src/locales/en/mystery-encounter.ts [hunk 3]
```diff
@@ -5,7 +5,7 @@ import { SimpleTranslationEntries } from "#app/interfaces/locales";
  * '$' will be treated as a new line for Message and Dialogue strings
  * '@d{<number>}' will add a time delay to text animation for Message and Dialogue strings
  *
- * '@ec{<token>}' will auto-inject the matching token value for the specified Encounter
+ * '{{<token>}}' will auto-inject the matching token value for the specified Encounter that is stored in dialogueTokens
```
Comment (flx-sta) [inline src/locales/en/mystery-encounter.ts:8]:
```suggestion
 * '{{<token>}}' will auto-inject the matching token value for the specified Encounter that is stored in dialogueTokens (see [i18next interpolations](https://www.i18next.com/translation-function/interpolation))
```

**Comment by ben-lear** [comment]
just added this
image

**Pull request merged**